x-pack/plugins/security_solution/server/endpoint/services/configuration_insights/helpers.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ElasticsearchClient } from '@kbn/core/server';
+
+import { DataStreamSpacesAdapter } from '@kbn/data-stream-adapter';
+
+import {
+  DATA_STREAM_PREFIX,
+  COMPONENT_TEMPLATE_NAME,
+  INDEX_TEMPLATE_NAME,
+  INGEST_PIPELINE_NAME,
+  TOTAL_FIELDS_LIMIT,
+} from './constants';
+import { securityConfigurationInsightsFieldMap } from './field_map_configurations';
+
+export function createDatastream(kibanaVersion: string): DataStreamSpacesAdapter {
+  const ds = new DataStreamSpacesAdapter(DATA_STREAM_PREFIX, {
+    kibanaVersion,
+    totalFieldsLimit: TOTAL_FIELDS_LIMIT,
+  });
+  ds.setComponentTemplate({
+    name: COMPONENT_TEMPLATE_NAME,
+    fieldMap: securityConfigurationInsightsFieldMap,
+  });
+  ds.setIndexTemplate({
+    name: INDEX_TEMPLATE_NAME,
+    componentTemplateRefs: [COMPONENT_TEMPLATE_NAME],
+    template: {
+      settings: {
+        default_pipeline: INGEST_PIPELINE_NAME,
+      },
+    },
+    hidden: true,
+  });
+  return ds;
+}
+
+export async function createPipeline(esClient: ElasticsearchClient): Promise<boolean> {
+  const response = await esClient.ingest.putPipeline({
+    id: INGEST_PIPELINE_NAME,
+    processors: [
+      // requires @elastic/elasticsearch release
```

Comment:
Fingerprint processor not defined in the current `8.15.1` release of `elastic-js` but it was added to the [repo about 2 weeks ago](https://github.com/elastic/elasticsearch-js/blob/2455dac4e5b1730c97d0fe4266612c88fc11e088/src/api/types.ts#L12882). We'll need a `elastic-js` release + version bump in kibana so we can uncomment this.